### ISSUE TITLE
Extension dev environment goes away after opening quick pick 

### ISSUE THREAD

**Issue by chrisdias** [Microsoft/vscode#42558]
MacOS
Version 1.20.0-insider (1.20.0-insider)
8cac1acd97466afc75c663dc1626d8562a9a3f3d
2018-01-30T17:15:07.352Z

Steps to Reproduce:

1. Get the Docker extension sources:

`git clone https://github.com/microsoft/vscode-docker`
`git fetch`
`git checkout chrisdias/42558`
`cd vscode-docker && npm install`
`code-insiders .`

2. Note that I've made a change to where `keytar` can be found in 

https://github.com/Microsoft/vscode-docker/blob/e5f94f114c027b7d49c3eee9756c1ba33f820509/explorer/models/registryRootNode.ts#L30

https://github.com/Microsoft/vscode-docker/blob/e5f94f114c027b7d49c3eee9756c1ba33f820509/explorer/utils/dockerHubUtils.ts#L82

3. Run the extension
4. Expand the `Registries` Explorer Node
5. Expand the `DockerHub` node
6. The prompt for Username will appear

Don't do anything, just wait!

Result: the dev env will disappear!

Not sure if this is related to the ASAR change for keytar, but this is how i can make it happen.

**Comment by alexandrudima** [comment]
I could not entirely reproduce, as the branch `chrisdias/42558` does not exist for me when I clone the source code. 

I did, however, patch master to replace the two occurences in `registryRootNode.ts` and `dockerHubUtils.ts`:
```
require(`${vscode.env.appRoot}/node_modules/keytar`);
=>
require(`${vscode.env.appRoot}/node_modules.asar.unpacked/keytar/build/Release/keytar.node`)
```

I have tried with today's insiders version (f2b21097100fc2fc0e84cc34f453c9a85a49a2bb) which has ASAR disabled and **could not reproduce anything unusual**.

I then proceeded to try with the insiders version 8adf87c03d78597e385ef787e8fb1ef564f8995b, which has ASAR enabled:
* [mac download](https://az764295.vo.msecnd.net/insider/8adf87c03d78597e385ef787e8fb1ef564f8995b/VSCode-darwin-insider.zip)
* [win download](https://az764295.vo.msecnd.net/insider/8adf87c03d78597e385ef787e8fb1ef564f8995b/VSCode-win32-x64-1.20.0-insider.zip)
* [linux download](https://az764295.vo.msecnd.net/insider/8adf87c03d78597e385ef787e8fb1ef564f8995b/code-insider-1.20.0-1517379630_amd64.tar.gz)

I can reproduce, and it appears that the window or the extension host is crashing.

**Comment by alexandrudima** [comment]
I have continued the investigation by soft-linking the modified extension as a user installed extension on my machine. Such that I can execute it outside of the debugging flow.

I could then see that the extension is causing a segmentation fault:
```
#
# Fatal error in ../../v8/src/objects.cc, line 3185
# Check failed: receiver->IsJSFunction().
#

==== C stack trace ===============================

    0   libnode.dylib                       0x0000000105b91563 v8::base::debug::DisableSignalStackDump() + 35
    1   libnode.dylib                       0x0000000105b8ef2d V8_Fatal + 221
    2   libnode.dylib                       0x00000001058d431d v8::internal::HeapNumber::HeapNumberPrint(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 1821
    3   libnode.dylib                       0x0000000105212015 node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*) + 53
    4   keytar.node                         0x00000001067d5239 Nan::Callback::Call_(v8::Isolate*, v8::Local<v8::Object>, int, v8::Local<v8::Value>*) const + 95
    5   keytar.node                         0x00000001067d49c8 GetPasswordWorker::HandleOKCallback() + 234
    6   keytar.node                         0x00000001067d4fe8 Nan::AsyncWorker::WorkComplete() + 58
    7   keytar.node                         0x00000001067d5c01 Nan::AsyncExecuteComplete(uv_work_s*) + 18
    8   libnode.dylib                       0x0000000105299aba uv_queue_work + 218
    9   libnode.dylib                       0x0000000105299998 uv_inet_pton + 2200
    10  libnode.dylib                       0x000000010529b60c uv_async_init + 556
    11  libnode.dylib                       0x000000010529b7e8 uv_async_send + 440
    12  libnode.dylib                       0x00000001052aa3f6 uv_free_interface_addresses + 8198
    13  libnode.dylib                       0x000000010529bba2 uv_run + 146
    14  Electron Framework                  0x00000001071bf2e5 atom::NodeMain(int, char**) + 965
    15  Electron Framework                  0x00000001071bb4bc AtomInitializeICUandStartNode + 300
    16  Code - Insiders Helper              0x0000000104fd5f1f main + 31
    17  libdyld.dylib                       0x00007fff64d0f115 start + 1
    18  ???                                 0x0000000000000003 0x0 + 3
```

i.e. the source code that you have written is invoking something on `keytar` in a way that `keytar` does not like and in a way that produces a segmentation fault.

IMHO, the error is in the loading of `keytar.node` directly, which is an implementation detail of the `keytar` node module.

**TL;DR** IMHO this is not a vscode problem, but rather a segmentation fault of the extension host caused by the extension...

**Comment by chrisdias** [comment]
as usual, another awesome analysis by @alexandrudima.  

makes perfect sense, thank you for looking into this.
